runbot_gitlab_minimal/models/runbot_repo.py

```diff
@@ -0,0 +1,64 @@
+# -*- coding: utf-8 -*-
+
+import os
+import logging
+import re
+import urllib
+
+import requests
+
+from openerp import models, fields
+
+_logger = logging.getLogger(__name__)
+
+
+class RunbotRepo(models.Model):
+    _inherit = "runbot.repo"
+
+    uses_gitlab = fields.Boolean(string='Use Gitlab', help='Use Gitlab')
+
+    def update_git(self, cr, uid, repo, context=None):
+        if os.path.isdir(os.path.join(repo.path, 'refs')) and repo.uses_gitlab:
+            repo.git(['fetch', '-p', 'origin',
+                      '+refs/merge-requests/*/head:refs/pull/*'])
+        return super(RunbotRepo, self).update_git(cr, uid, repo,
+                                                  context=context)
+
+    def github(self, cr, uid, ids, url, payload=None, ignore_errors=False,
+               context=None):
+        for repo in self.browse(cr, uid, ids, context=context):
+            if not repo.token:
+                continue
+            if not repo.uses_gitlab:
+                return (super(RunbotRepo, self)
+                        .github(cr, uid, [repo.id], payload=payload,
+                                ignore_errors=ignore_errors, context=context))
+            else:
+                try:
+                    match_object = re.search('([^/]+)/([^/]+)/([^/.]+(.git)?)',
+                                             repo.base)
+                    if match_object:
+                        project_name = (match_object.group(2) + '/' +
+                                        match_object.group(3))
+                        url = url.replace(':owner', match_object.group(2))
+                        url = url.replace(':repo', match_object.group(3))
+                        url = 'https://%s/api/v3%s' % (match_object.group(1),
+                                                       url)
+                        url = url.replace('/repos/', '/projects/')
```

Comment:
Check if `/repos/:owner/:repo/pulls/%s % pull_number` works for gitlab or required a replace

https://github.com/odoo/odoo-extra/blob/master/runbot/runbot.py#L490